extensions/amp-mega-menu/0.1/test/test-amp-mega-menu.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../amp-mega-menu';
+import {Keys} from '../../../../src/utils/key-codes';
+import {toggleExperiment} from '../../../../src/experiments';
+import {tryFocus} from '../../../../src/dom';
+
+describes.realWin(
+  'amp-mega-menu',
+  {
+    amp: {
+      extensions: ['amp-mega-menu'],
+    },
+  },
+  env => {
+    let win, doc, element;
+
+    beforeEach(() => {
+      win = env.win;
+      doc = win.document;
+
+      element = getAmpMegaMenu();
+      doc.body.appendChild(element);
+
+      toggleExperiment(win, 'amp-mega-menu', true);
+    });
+
+    function getAmpMegaMenu() {
+      const element = doc.createElement('amp-mega-menu');
+      element.setAttribute('layout', 'fixed-height');
+      element.setAttribute('height', '10');
+      const nav = doc.createElement('nav');
+      const ul = doc.createElement('ul');
+      element.appendChild(nav);
+      nav.appendChild(ul);
+
+      const items = [1, 2, 3].map(i => {
+        return (
+          `<button id="heading${i}">Menu Item ${i}</button>` +
+          `<div id="content${i}" role="dialog">Loreum ipsum</div>`
+        );
+      });
+
+      for (let i = 0; i < items.length; i++) {
+        const li = doc.createElement('li');
+        li.innerHTML = items[i];
```

Comment:
We can use `htmlFor` instead of directly setting `innerHTML`. =)

Reply:
it turns out `htmlFor` cannot take in a template with dynamic `${expression}`, so I removed the for loop and just wrote out the items; this seems more readable anyway imo.